documentation/TESTING_PLAN.md

```diff
@@ -16,8 +16,28 @@ In general, we're approaching testing bottom-up, with each unit function tested
 
 ### Unit-level tests
 
-- TBD 10
+- undo() should load the previous game state.
+- reset() should load the current level to its original state.
+- move(U) should allow the player to move up if it's a valid move.
+- move(D) should allow player to move down.
+- move(L) should allow player to move left.
+- move(R) should allow player to move right.
+- player should not be able to pass through impassable walls when move(U).
+- player should not be able to pass through impassable walls when move(D).
+- player should not be able to pass through impassable walls when move(L).
+- player should not be able to pass through impassable walls when move(R).
+- change_level(uint level_number) should allow user to switch levels.
+- moves() should return the number of moves made on the current board.
+- solved() should return true when all the boxes are on top of goal cells.
```

Comment:
Maybe we should throw in some box pushing tests -- the 4 directions for each push are good, but also somewhat redundant as far as the requirements for 10 tests go.

Good function names here, but I'd probably keep the function call/code out of it so it's a bit more high-level -- maybe "should undo a move", "should disallow moving into a wall from the right", "should detect a solved board", "should detect an unsolved board", etc.